doc/guide/authentication.xml

```diff
@@ -99,4 +99,75 @@
 
     </section>
   </section>
+
+  <section id="twofa-auth">
+    <title>Enable two-factor authentication</title>
+
+    <para>As with any other <code>pam</code> based service that supports it,
+      you can harden your Cockpit authentication by adding TOTP/HOTP second factor authentication.
+      This can be achieved by installing the <code>google-authenticator</code> package
+      and configuring <code>pam</code> to use it with Cockpit.</para>
+
+          <para>Fedora/CentOS</para>
+<programlisting>
+sudo dnf install google-authenticator
+</programlisting>
+
+          <para>openSUSE</para>
+<programlisting>
+sudo zypper install google-authenticator-libpam
+</programlisting>
+
+          <para>Debian based OS</para>
+<programlisting>
+sudo apt install libpam-google-authenticator
+</programlisting>
+
+          <para>Arch Linux</para>
+<programlisting>
+sudo pacman -S libpam-google-authenticator
+</programlisting>
+
+    <para>As the user you want to set up two-factor auth for, configure
+      <code>google-authenticator</code> as per its <ulink url="https://github.com/google/google-authenticator-libpam">documentation</ulink>.
+    </para>
+
+    <note><title>Configuration file path</title>
+      <para>By default, the configuration file will be created as
+        <code>$HOME/.google_authenticator</code>. You may want
+        to move it in a more convenient and standard place, such as
+        <code>$HOME/.config/google-authenticator/google_authenticator</code>.
+      </para>
+    </note>
+
+    <para>Then, edit the <filename>/etc/pam.d/cockpit</filename> file and
```

Comment:
Doesn't this need to change `pam_unix` from `sufficient` to `required` in `/etc/pam.d/system-auth`? At least that's what I need to do on Fedora.

Reply:
On my Fedora Server instance, all I had to do was editing `/etc/pam.d/cockpit`/. It was also the case on Debian.